daemon/api.go

```diff
@@ -748,22 +717,27 @@ func postSnap(c *Command, r *http.Request) Response {
 		return BadRequest("unknown action %s", inst.Action)
 	}
 
-	return AsyncResponse(c.d.AddTask(func() interface{} {
-		lock, err := lockfile.Lock(dirs.SnapLockFile, true)
-		if err != nil {
-			return err
-		}
-		defer lock.Unlock()
-		return f()
-	}).Map(route))
+	chg, err := f()
+	if err != nil {
+		return InternalError("can't %s %q: %v", inst.Action, inst.pkg, err)
+	}
+
+	url, err := route.URL("id", chg.ID())
+	if err != nil {
+		return InternalError("route can't build URL for change: %v", err)
+	}
+
+	return AsyncResponse(map[string]interface{}{
+		"resource": url.String(),
```

Comment:
I think this would be better as a top-level `change` field in the Response, but I can do that on my side, together with source, etc.